src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -365,7 +365,7 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                 }
                 catch (Exception e) when (ExceptionHelper.BeforeCatch(e, Logger, reportOnlyCorrupting: true))
                 {
-                    if (this.ProcessState == MICore.ProcessState.Exited)
+                    if (_terminating)
```

Comment:
It safer that way. 
It worth noticing that for all the scenarios I tested, this code can only be reached when stop debugging and debugger send "kill" command, GDB will have output of "signal-received" and debugger in turn tries to handle that as a break event. When debuggee run to exit we don't execute this code path.